core/editor/src/document/layer_panel.rs

```diff
@@ -0,0 +1,89 @@
+use crate::{consts::ROTATE_SNAP_INTERVAL, frontend::layer_panel::*};
+use document_core::{
+	layers::{Layer, LayerData as DocumentLayerData},
+	LayerId,
+};
+use glam::{DAffine2, DVec2};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Copy)]
+pub struct LayerData {
+	pub selected: bool,
+	pub expanded: bool,
+	pub translation: DVec2,
+	pub rotation: f64,
+	pub snap_rotate: bool,
+	pub scale: f64,
+}
+
+impl LayerData {
+	pub fn new(expanded: bool) -> LayerData {
+		LayerData {
+			selected: false,
+			expanded,
+			translation: DVec2::ZERO,
+			rotation: 0.,
+			snap_rotate: false,
+			scale: 1.,
+		}
+	}
+	pub fn snapped_angle(&self) -> f64 {
```

Comment:
I think we should be sure to always have blank lines between functions in `impl` blocks. Assuming you agree, check for this elsewhere in this CL.